lib/core/base/audit.js

```diff
@@ -672,6 +520,158 @@ class Audit {
   }
 }
 
+function getDefaultOrigin() {
+  // @see https://html.spec.whatwg.org/multipage/webappapis.html#dom-origin-dev
+  // window.origin does not exist in ie11
+  // prevent origin default "null" string on CDP `Page.setDocumentContent` https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-setDocumentContent
+  if (window.origin && window.origin !== 'null') {
+    return window.origin;
+  }
+  // window.location does not exist in node when we run the build
+  if (
+    window.location &&
+    window.location.origin &&
+    window.location.origin !== 'null'
+  ) {
+    return window.location.origin;
+  }
+}
+
+/*eslint no-unused-vars: 0*/
+function getDefaultConfiguration(audit) {
+  var config;
+  if (audit) {
+    config = clone(audit);
+    // Commons are configured into axe like everything else,
+    // however because things go funky if we have multiple commons objects
+    // we're not using the copy of that.
+    config.commons = audit.commons;
+  } else {
+    config = {};
+  }
+
+  config.reporter = config.reporter || null;
+  config.noHtml = config.noHtml || false;
+
+  if (!config.allowedOrigins) {
+    const defaultOrigin = getDefaultOrigin();
+    config.allowedOrigins = defaultOrigin ? [defaultOrigin] : [];
+  }
+
+  config.rules = config.rules || [];
+  config.checks = config.checks || [];
+  config.data = { checks: {}, rules: {}, ...config.data };
+  return config;
+}
+
+function unpackToObject(collection, audit, method) {
+  var i, l;
+  for (i = 0, l = collection.length; i < l; i++) {
+    audit[method](collection[i]);
+  }
+}
+
+/**
+ * Merge two check locales (a, b), favoring `b`.
+ *
+ * Both locale `a` and the returned shape resemble:
+ *
+ *    {
+ *      impact: string,
+ *      messages: {
+ *        pass: string | function,
+ *        fail: string | function,
+ *        incomplete: string | {
+ *          [key: string]: string | function
+ *        }
+ *      }
+ *    }
+ *
+ * Locale `b` follows the `axe.CheckLocale` shape and resembles:
+ *
+ *    {
+ *      pass: string,
+ *      fail: string,
+ *      incomplete: string | { [key: string]: string }
+ *    }
+ */
+
+const mergeCheckLocale = (a, b) => {
```

Comment:
Just curious - when is a function declaration preferred over a function expression? Also why is one preferred over the other?

Reply:
As far as I know we have no standard for when to use either. Mostly it's been up to the person writing the code.

Reply:
Both functions created using expressions and declarations have a name.

```
Welcome to Node.js v16.14.0.
Type ".help" for more information.
> function foo(){}
undefined
> foo.name
'foo'
> const bar = () => {}
undefined
> bar.name
'bar'
>
```